### PR TITLE
Added support of 1x1x1xN input for parseYolo

### PR DESCRIPTION
**Pull request by rgarnov**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [x] There is reference to original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

```
build_gapi_standalone:Linux x64=ade-0.1.1f
build_gapi_standalone:Win64=ade-0.1.1f
build_gapi_standalone:Mac=ade-0.1.1f
build_gapi_standalone:Linux x64 Debug=ade-0.1.1f
```

### CODE REVIEW

File: modules/gapi/src/backends/cpu/gnnparsers.cpp [hunk 1]
```diff
@@ -246,6 +246,33 @@ void parseSSD(const cv::Mat&  in_ssd_result,
     }
 }
 
+static void checkYoloDims(const MatSize& d) {
+    // Accept 1x1x1xN, 1x1xN, 1xN,
+    // 1x13x13xN and 13x13xN
+    GAPI_Assert(d.dims() >= 1);
+    const auto checkOneDimCase = [&](){
+        GAPI_Assert(d[d.dims()-1]%(5*13*13) == 0);
+        for (int i = 0; i < d.dims()-2; i++) {
```
Comment (AsyaPronina) [inline modules/gapi/src/backends/cpu/gnnparsers.cpp:255]: ``d.dims() - 1`` should be as a limit

Comment (AsyaPronina) [reply modules/gapi/src/backends/cpu/gnnparsers.cpp:255]: Please check 1x1x3x21125, I think it should be acceptable with the current algorithm

Comment (rgarnov) [reply modules/gapi/src/backends/cpu/gnnparsers.cpp:255]: That's a nice catch, fixed, thanks!

**Review by AsyaPronina — APPROVED** [review]

File: modules/gapi/src/backends/cpu/gnnparsers.cpp [hunk 2]
```diff
@@ -257,11 +257,10 @@ void parseYolo(const cv::Mat&  in_yolo_result,
     const auto& dims = in_yolo_result.size;
     GAPI_Assert(dims.dims() == 4);
```
[the remark this answers is not in the document]
Comment (rgarnov) [reply modules/gapi/src/backends/cpu/gnnparsers.cpp]: Looks like you are referring to the old version

File: modules/gapi/src/backends/cpu/gnnparsers.cpp [hunk 3]
```diff
@@ -246,6 +246,33 @@ void parseSSD(const cv::Mat&  in_ssd_result,
...
+    if (d.dims() >= 3) {
+        if (d[d.dims()-2] == 13) {
+            GAPI_Assert(d[d.dims()-1]%5 == 0);
+            GAPI_Assert(d[d.dims()-2] == 13);
+            GAPI_Assert(d[d.dims()-3] == 13);
+            for (int i = 0; i < d.dims()-3; i++) {
+                GAPI_Assert(d[i] == 1);
+            }
+        } else {
+            checkOneDimCase();
```
Comment (alalek) [inline modules/gapi/src/backends/cpu/gnnparsers.cpp:269]: You can avoid lambda (looks bad in assertion messages) and expand its content below if you put `return` after `for`.

Also you may want to cache `d.dims()`: `const int d = d.dims()`

**Comment by alalek** [comment]
@dmatveev If you want to delegate final merge "approval" then please "assign" PR on corresponding reviewer (you can do even before receiving "approved" status).
In that case I would not wait for your approval for PR.

File: modules/gapi/src/backends/cpu/gnnparsers.cpp (hunk 3 above)
Comment (rgarnov) [reply modules/gapi/src/backends/cpu/gnnparsers.cpp]: Fixed, thanks!

**Pull request merged**